src/core/reactor.ts

```diff
@@ -1608,6 +1609,13 @@ export class App extends Reactor {
      */
     private _reactorsToRemove = new Array<Reactor>();
 
+    /**
+     * Stores whether the last received Tag was provisional.
+     * Every federate starts out assuming that it has been granted a PTAG
+     * at the start time, or if it has no upstream federates, a TAG.
+     */
+    protected _isLastTagProvisional: boolean = false;
```

Comment:
Rename this to isCurrentTagProvisional (also suggest change in reactor-c).

Reply:
The field is `is_last_TAG_provisional` in reactor-c. Notice the capitalized TAG, which stands for Tag Advance Grant. This field indicates whether or not the last TAG message received from the RTI was a PTAG. This field is not related to the current tag.

Reply:
This is super confusing. I understand why @hokeun would write `_isLastTagProvisional` instead of the correct spelling `_isLastTAGProvisional` which is only marginally different. The adoption of `TAG` as a message type is really unfortunate... Why do we need the `T`? Isn't "Advance Grant" clear enough? Are there any other things granted? Regardless, a less error-prone name for the field could be `_isLastGrantProvisional`.

Reply:
TAG is a recursive acronym for Tag Advance Grant :) 
It is a descendant of `TimeAdvancedGrant` in HLA.

With that said, I don't personally have a problem with renaming it.
I think `advance grant` is a bit vague since it is not clear what is meant to advance.

Nonetheless, I think in the context of this particular flag, the documentation is very elaborate:
```C
/**
 * Indicates whether the last TAG received is provisional or an ordinary
 * TAG. 
 * If the last TAG has been provisional, network control reactions must be inserted.
 * This variable should only be accessed while holding the mutex lock.
 */
```

Reply:
I understand that `TAG` is a recursive acronym, which is nice, but it's also confusing because it is easily confused with the notion of a tag. For not, I think the most pragmatic thing would be to rename the field to `_isLastGrantProvisional` (and, more generally, try to avoid the term TAG in method names or fields).

Reply:
Maybe `_isLastRTIGrantProvisional`?

This appears to be in `reactor.ts` not `federation.ts`, which might require extra context in the name. I'm not an expert in naming conventions in TS though.

Reply:
It's not an RTI grant though -- it's a Tag Advance Grant. Leaving off the Tag part doesn't introduce any ambiguity provided there are no other kinds of grants happening (but if there were the name you suggest would be equally ambiguous).

Reply:
Thank you @Soroosh129 and @lhstrh ! I renamed this to _isLastTAGProvisional in https://github.com/lf-lang/reactor-ts/pull/91/commits/a0eb024cfe944b073cebdb739e0e52735dde3e35.
This may not be the best, but follows the naming in reactor-c. I also updated the comments.